docs/immersiveModal.md

```diff
@@ -0,0 +1,89 @@
+# ImmersiveModal
+### Usage
+
+```jsx
+import React from 'react';
+
+import { ImmersiveModal, Button } from 'radiance-ui';
+
+const HeaderImage = () => (
+  <div>
+    This is a placeholder for an optional header image.
+  </div>
+);
+
+class DefaultImmersiveModal extends React.Component {
+  state = {
+    defaultIsOpen: false,
+    headerIsOpen: false,
+  };
+
+  onOpenDefaultModal = () => this.setState({ defaultIsOpen: true });
+
+  onOpenHeaderModal = () => this.setState({ headerIsOpen: true });
+
+  onClose = () => this.setState({
+    defaultIsOpen: false,
+    headerIsOpen: false,
+  });
+
+  render() {
+    const { defaultIsOpen, headerIsOpen } = this.state;
+
+    return (
+      <div>
+        <Button onClick={this.onOpenDefaultModal}>Open ImmersiveModal</Button>
+        {defaultIsOpen && (
+          <ImmersiveModal onClose={this.onClose}>
+            <ImmersiveModal.Title>This is styled with ImmersiveModalTitle</ImmersiveModal.Title>
```

Comment:
probably want to update the text to say "This is styled with `ImmersiveModal.Title`". Similar comment for the rest of the file